x-pack/filebeat/input/httpjson/request.go

```diff
@@ -672,10 +671,16 @@ func (r *requester) processChainPaginationEvents(ctx context.Context, trCtx *tra
 			}
 
 			// collect data from new urls
-			httpResp, err = rf.collectResponse(ctx, chainTrCtx, r)
+			httpResp, err := rf.collectResponse(ctx, chainTrCtx, r)
 			if err != nil {
 				return -1, fmt.Errorf("failed to collect response: %w", err)
 			}
+			defer func() {
+				if httpResp != nil && httpResp.Body != nil {
+					httpResp.Body.Close()
+				}
+			}()
```

Comment:
Oh, right. Now that I look more closely, I see: https://github.com/elastic/beats/blob/df064b8150517ff085b32f617af0be56c2b0a908/x-pack/filebeat/input/httpjson/request.go#L503 in `getIdsFromResponses`, which we can get rid of since we are already draining the body and closing it with `drainBody`?

I'll remove this as well.